### PR TITLE
Selectable ship menu

### PR DESCRIPTION
**Pull request by squid-ai**
Rendering of two ships working
Button functionality does not yet work for the ships
Space shooter game not playable - will not be loaded

### CODE REVIEW

File: src/seagulls/examples/space_shooter/_selectable_ship_menu.py [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+    GameControls,
+    GameObject,
+    GameObjectsCollection,
+    IGameScene,
+    Rect,
+    Surface,
+    SurfaceRenderer
+)
+from seagulls.examples import ISetActiveScene
+from seagulls.examples.space_shooter._active_ship_client import ISetActiveShip
```
Comment (Zeelot) [inline src/seagulls/examples/space_shooter/_selectable_ship_menu.py:22]: try making this a relative import

File: src/seagulls/examples/space_shooter/_active_ship_client.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
+from abc import ABC, abstractmethod
+from typing import Callable
+
+from seagulls.examples.space_shooter._selectable_ship_menu import IShip
```
Comment (Zeelot) [inline src/seagulls/examples/space_shooter/_active_ship_client.py:4]: I think this one has to be relative too!

File: src/seagulls/examples/space_shooter/_ship.py [hunk 3]
```diff
@@ -13,6 +13,7 @@
     Surface,
     Vector2
 )
+from seagulls.examples.space_shooter._active_ship_client import IProvideActiveShip
```
Comment (Zeelot) [inline src/seagulls/examples/space_shooter/_ship.py:16]: and this one

File: src/seagulls/examples/space_shooter/_space_collisions.py [hunk 4]
```diff
@@ -1,10 +1,11 @@
 import logging
-from typing import Callable
 from pathlib import Path
+from typing import Callable
+
 from pygame.font import Font
 
 from seagulls.engine import GameObject, Surface
-from seagulls.examples.space_shooter import Ship, AsteroidField
+from seagulls.examples.space_shooter import AsteroidField, Ship
```
Comment (Zeelot) [inline src/seagulls/examples/space_shooter/_space_collisions.py:8]: this one too :)

File: src/seagulls/examples/space_shooter/_selectable_ship_menu.py (hunk 1 above)
Comment (squid-ai) [reply src/seagulls/examples/space_shooter/_selectable_ship_menu.py]: updated

File: src/seagulls/examples/space_shooter/_ship.py (hunk 3 above)
Comment (squid-ai) [reply src/seagulls/examples/space_shooter/_ship.py]: updated

**Review by Zeelot — APPROVED** [review]

**Pull request merged**